src/OrchardCore.Modules/OrchardCore.Users/Localization/UserLocalizationStartup.cs

```diff
@@ -0,0 +1,23 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.Modules;
+using OrchardCore.Users.Localization.Drivers;
+using OrchardCore.Users.Localization.Providers;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+
+namespace OrchardCore.Users.Localization;
+
+[Feature("OrchardCore.Users.Localization")]
+public class UserLocalizationStartup : StartupBase
+{
+    public override void ConfigureServices(IServiceCollection services)
+    {
+        services.AddScoped<IDisplayDriver<User>, UserLocalizationDisplayDriver>();
+        services.AddScoped<IUserClaimsProvider, UserLocalizationClaimsProvider>();
+
+        services.Configure<RequestLocalizationOptions>(options =>
+            options.AddInitialRequestCultureProvider(new UserLocalizationRequestCultureProvider()));
```

Comment:
Okay no problem, it's an extension that just does what you were doing explicitly ;)

Reply:
Sure, we should utilize the localization stuff as much as we can (DRY)